diff --git a/Makefile b/Makefile
index 334847d..c9fb1e5 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-CFLAGS = -Wall -O3 -funroll-loops -I/usr/include
+CFLAGS = -Wall -O3 -funroll-loops
 OCAMLC = ocamlc
 OCAMLOPT = ocamlopt
 OCAMLMKLIB = ocamlmklib
diff --git a/sha1_stubs.c b/sha1_stubs.c
index e0260bc..309ae5c 100644
--- a/sha1_stubs.c
+++ b/sha1_stubs.c
@@ -16,6 +16,7 @@
 #define _GNU_SOURCE
 #include <unistd.h>
 #include <fcntl.h>
+#include <string.h>
 #include "sha1.h"
 
 static inline int sha1_file(char *filename, sha1_digest *digest)
@@ -46,6 +47,7 @@ static inline int sha1_file(char *filename, sha1_digest *digest)
 #include <caml/fail.h>
 #include <caml/bigarray.h>
 #include <caml/threads.h>
+#define Bigarray_val Caml_ba_array_val
 
 #define GET_CTX_STRUCT(a) ((struct sha1_ctx *) a)
 
diff --git a/sha256_stubs.c b/sha256_stubs.c
index 303171d..9ab4bc1 100644
--- a/sha256_stubs.c
+++ b/sha256_stubs.c
@@ -16,6 +16,7 @@
 #define _GNU_SOURCE
 #include <unistd.h>
 #include <fcntl.h>
+#include <string.h>
 #include "sha256.h"
 
 static inline int sha256_file(char *filename, sha256_digest *digest)
@@ -46,6 +47,7 @@ static inline int sha256_file(char *filename, sha256_digest *digest)
 #include <caml/fail.h>
 #include <caml/bigarray.h>
 #include <caml/threads.h>
+#define Bigarray_val Caml_ba_array_val
 
 #define GET_CTX_STRUCT(a) ((struct sha256_ctx *) a)
 
diff --git a/sha512_stubs.c b/sha512_stubs.c
index 2a7a071..1a75204 100644
--- a/sha512_stubs.c
+++ b/sha512_stubs.c
@@ -16,6 +16,7 @@
 #define _GNU_SOURCE
 #include <unistd.h>
 #include <fcntl.h>
+#include <string.h>
 #include "sha512.h"
 
 static inline int sha512_file(char *filename, sha512_digest *digest)
@@ -46,6 +47,7 @@ static inline int sha512_file(char *filename, sha512_digest *digest)
 #include <caml/fail.h>
 #include <caml/bigarray.h>
 #include <caml/threads.h>
+#define Bigarray_val Caml_ba_array_val
 
 #define GET_CTX_STRUCT(a) ((struct sha512_ctx *) a)
 
